### PR TITLE
fix: improve test coverage and e2e reliability

### PR DESCRIPTION
**Pull request by lcqiw**
### Code Quality
- Replaces hardcoded `"90 days"` in tests with `constant.DefaultAPIKeyMaxExpirationDays`

### Test Coverage
- Adds 5 new test cases for `GetMaxExpirationDays` covering nil config, zero, custom, and negative values
- Verifies integration with `CreateAPIKey` method

### E2E Improvements
- Fixes base64url padding calculation in `_decode_jwt_payload`
- Replaces `time.sleep(3)` with a polling helper (`_wait_for_api_key_revocation`) for deterministic revocation checks
- Tightens HTTP 500 handling to require `AUTH_FAILURE` confirmation in response body

### CODE REVIEW

**Review by liangwen12year — COMMENTED** [review]
A few observations:

**1. Subscription filter — subtle behavior change** — The store changes restructure the `nil`+empty check:
   ```go
   // Before (main):
   if filters.Subscription != nil && *filters.Subscription != "" {
       sub := strings.TrimSpace(*filters.Subscription)
       // filters by sub — even if TrimSpace reduced it to ""
   }
   
   // After (this PR):
   if filters.Subscription != nil {
       sub := strings.TrimSpace(*filters.Subscription)
       if sub != "" { /* filter */ }
   }
   ```
   If `*filters.Subscription` is whitespace-only (e.g. `"  "`), the old code would filter for `Subscription == ""` (effectively returning nothing). The new code skips filtering entirely (returning all keys). Edge case, but it's a behavioral change worth calling out.

**2. Tests hardcode `90`** — e.g. `assert.Equal(t, 90, result)` instead of using `constant.DefaultAPIKeyMaxExpirationDays`. If the default ever changes, these tests silently break rather than adapting.

**3. E2E changes look good** — Replacing `time.sleep(3)` with a polling helper and fixing the base64 padding calculation are both improvements.

File: maas-api/internal/api_keys/service.go [hunk 1]
```diff
@@ -59,6 +59,7 @@ func NewServiceWithLogger(store MetadataStore, cfg *config.Config, sub Subscript
 	}
 }
 
+
```
Comment (jrhyness) [inline maas-api/internal/api_keys/service.go]: Please remove the extra line and  update the PR description to match the final contents of the changes.

**Review by jrhyness — CHANGES_REQUESTED** [review]

File: test/e2e/tests/test_external_oidc.py [hunk 2]
```diff
@@ -180,6 +180,28 @@ def _create_oidc_api_key(
...
+    while time.time() - start < timeout:
+        response = requests.get(
+            f"{maas_api_base_url}/v1/models",
+            headers={"Authorization": f"Bearer {api_key}", "Content-Type": "application/json"},
+            timeout=30,
+            verify=TLS_VERIFY,
+        )
+        if response.status_code in (401, 403, 500):
+            log.info("Revocation enforced after %.1fs (status=%d)", time.time() - start, response.status_code)
+            return
```
[the remark this answers is not in the document]
Comment (jrhyness) [reply test/e2e/tests/test_external_oidc.py:212]: Agreed

**Review by jrhyness — APPROVED** [review]
lgtm
